sqlmesh/core/context_diff.py

```diff
@@ -204,6 +207,11 @@ def promotable_models(self) -> t.Set[str]:
             *self.modified_snapshots,
         } - self.removed
 
+    @property
+    def unpromoted_models(self) -> t.Set[str]:
+        """The set of model names that have have not yet been promoted in the target environment."""
+        return self.existing - self.previously_promoted_model_names
```

Comment:
I don't think we need `existing`. The combination of `added` and `modified_snapshots` should give us all we need as far as I can tell.

Reply:
Actually, no, isn't it just `snapshots`?

Reply:
Good call